Week 1 653380198-5 นางสาวทักษพร มูลมณี

Lab 1.1: Environment Setup & First Script

In [ ]:
print("Hello","World!")

Hello World!


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


2. Declare and print variables:


In [ ]:
Age = 22
GPA = 1.87
Favorite_Color = "Blue"
Student = True

print("Age = ",Age ,type(Age))
print("GPA = ",GPA,type(GPA))
print("Favorite_Color = ",Favorite_Color,type(Favorite_Color))
print("Student = ",Student,type(Student))

Age =  22 <class 'int'>
GPA =  1.87 <class 'float'>
Favorite_Color =  Blue <class 'str'>
Student =  True <class 'bool'>


3. Perform simple arithmetic:


In [ ]:
num1 = 8
num2 = 5
sum = num1 + num2
print("Sum = ",sum)

Sum =  13


4. String concatenation and f-strings:


In [ ]:
First_name = "Tagsaporn"
Last_name = "Moonmanee"
Full_name = First_name + " " + Last_name
print(f"Hello,{Full_name}")

Hello,Tagsaporn Moonmanee


5. User input and type conversion:


In [ ]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 72
type of A is :  <class 'str'>
72  + 5 =  77


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ:สามารถวิเคราะห์ข้อมูลเบื้องต้น การเขียนสคริปต์สำหรับเกม และการพัฒนาเว็บไซต์หรืออื่นๆ

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: 13.5
<class 'float'>

ทำไมถึงเป็นtype?
คำตอบ: x = 10 เป็น int (จำนวนเต็ม ไม่มีทศนิยม)  
       y = 3.5 เป็น float (มีจุดทศนิยม)
       เมื่อนำสองตัวแปรที่มี type ต่างกันมาบวกกัน (x+y) ใช้กลไกเรียกว่า type coercion คือการแปลงประเภทข้อมูลให้ตรงกันโดยอัตโนมัติ โดยไม่ต้องให้เราสั่งแปลงเอง หรือสรุปง่ายๆก็คือ int + float = float นั้นเอง

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old.
<class 'str'>

ทำไมถึงเป็นtype...?
คำตอบ: name + "is" + age +"year old" = "Alice is 30 years old." แต่แรกเดิม name เป็น string และ age ถ้าดูดีๆเป็น string แต่แรกเพราะมี "30" ไม่ได้เป็น int การที่เอาข้อความมา + กันไม่ใช่บวกตัวเลขแต่เป็นการรวมข้อมูลให้เป็นก้อนเดียวกัน

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์:False
<class 'bool'>

ทำไมถึงเป็นtype...?
คำตอบ: is_sunny มีค่าเป็น boolean (bool) คือมีค่าได้เพียง True or False แน่นอนว่า is_raining มีค่าเป็น boolean และ and เป็นตรรกะ จะเป็นการคืนค่าและจะ True ต่อเมื่อทั้งสองฝั่งเป็น True ทั้งคู่ หากมี False ฝั่งใดฝั่งนึงจะเป็น Fales ทันที

In [ ]:
# Puzzler 4
# What happens here?
#num_str = input("Enter a number: ")
#result = num_str + 5 #แปลงเป็นค่า int ก่อนค่อยบวก
#print(result)

ได้ผลลัพธ์:TypeError: can only concatenate str (not "int") to str

ทำไมถึงเป็นtype...?
คำตอบ:เพราะถ้าเกิดการรันจริงจะเกิดการ Error ถึงเราจะใส่เลข 5 ก็เกิดการ Error เหมือนเดิม เพราะ num_str เป็นคลาส string ดังนั้นถ้าเอา num_str (string) มา + 5(int) จึงทำให้ python ไม่อนุญาติให้นำมาบวกกันโดยตรงเพราะจะเกิด TypeError ขึ้น python มองว่า string + int มันกำกวมและไม่ปลอดภัย